### PR TITLE
Validate: undefined and null are okay for optional fields

### PR DESCRIPTION
**Pull request by DaveStein**

### CODE REVIEW

File: util/validate.js [hunk 1]
```diff
@@ -214,19 +214,19 @@ define(function() {
...
-      validate.message = 'You must enter a value';
-      return false;
-    }
-
     delete validate.message;
     rules = rules ? commaSplit(rules) : [];
 
     // When optional field with no value
-    if (rules.indexOf('required') === -1 && body === '') {
+    if (rules.indexOf('required') === -1 && !body) {
```
Comment (Aintaer) [inline util/validate.js]: This is going to fail `body === 0`

File: test/specs/util/validate.js [hunk 2]
```diff
@@ -82,13 +82,17 @@ define(['util/validate'], function(validate) {
       tests[key].bad.forEach(bad, validator);
     }
 
+    function notNullUndefined(val) {
+      return !(val === null || val === undefined);
```
Comment (Aintaer) [inline test/specs/util/validate.js]: This is `val != null` Weak check against null is an accepted use of weak checks for `null` and `undefined`

Comment (DaveStein) [inline test/specs/util/validate.js]: So should I rewrite as what... or keep?

Comment (Aintaer) [inline test/specs/util/validate.js]: so just `return val != null;`

File: util/validate.js [hunk 3]
```diff
@@ -214,14 +214,14 @@ define(function() {
...
-    if (typeof body !== 'string') {
-      validate.message = 'You must enter a value';
-      return false;
-    }
-
     delete validate.message;
     rules = rules ? commaSplit(rules) : [];
 
+    // /.+/.test(null) passes as true for required check
+    if (body === null || body === undefined) {
```
Comment (Aintaer) [inline util/validate.js]: This is `body == null`

**Comment by Aintaer** [comment]
:accept:

**Pull request merged**